src/main/java/org/springframework/data/couchbase/core/convert/CouchbasePropertyValueConverterFactory.java

```diff
@@ -50,35 +52,48 @@ public <DV, SV, P extends ValueConversionContext<?>> PropertyValueConverter<DV,
 		if (valueConverter != null) {
 			return valueConverter;
 		}
+
 		Encrypted encryptedAnn = property.findAnnotation(Encrypted.class);
 		if (encryptedAnn != null) {
 			Class cryptoConverterClass = CryptoConverter.class;
 			return getConverter((Class<PropertyValueConverter<DV, SV, P>>) cryptoConverterClass);
-		} else {
+		}
+
+		Class<?> type = property.getActualType();
+		if (!Enum.class.isAssignableFrom(type)) {
 			return null;
 		}
+
+		// for Enums only
+
+		for (Method m : type.getDeclaredMethods()) {
+			if (m.getAnnotation(JsonValue.class) != null) {
+				Class jsonValueConverterClass = JsonValueConverter.class;
+				return getConverter((Class<PropertyValueConverter<DV, SV, P>>) jsonValueConverterClass);
+			}
+		}
```

Comment:
look for a JsonValue annotation. 
@dnault -  Should this also look for a shaded JsonValue annotation?